tests/extmod/vfs_posix.py

```diff
@@ -51,6 +52,26 @@
 print(f.read())
 f.close()
 
+# file finaliser, also see vfs_fat_finaliser.py
+names = [temp_dir + "/x%d" % i for i in range(4)]
+
+for i in range(1024):  # move GC head forwards by allocating a lot of single blocks
+    []
+
+
+def write_files_without_closing():
+    for n in names:
+        open(n, "w").write(n)
+    sorted(list(range(128)), key=lambda x: x)  # use up Python and C stack so f is really gone
+
+
+write_files_without_closing()
+gc.collect()
+
+for n in names:
+    print("deleting", n)
+    uos.remove(n)  # fails if file would still be open
```

Comment:
I believe some filesystems do allow removing open file handles and just doesn't actually delete the blocks until the handle is closed? So on some platforms you may be able to get a false pass on this?

Regardless, this test will be run on a number of platforms in CI, at least one of which should work as intended.

Reply:
Good point. I'd rather have this verified properly though. Would calling remove followed by listing the directory and checking the files aren't there catch everything?

Reply:
I'm not certain which filesystems behave this way to test (if I'm even recalling correctly in the first place), but I think the remove command will take them out of a file listing and the file will appear completely deleted - however the "disk space available" on the drive won't go up until the file is closed. 

I've seen this behaviour on a Linux computer after I've deleted large log files to make space on the drive, but it doesn't work until I restart the service. I no longer have the particular server I'm thinking of, but it was probably an ext4 filesystem.

Reply:
The linux man pages say 

> If the name was the last link to a file but any processes still
> have the file open, the file will remain in existence until the
> last file descriptor referring to it is closed.

so this seems fairly standard.

What does seem to work is opening a new file and checking `fileno`. And if I interpret

> The file descriptor returned by a successful `open` call will be the lowest-numbered file descriptor not
> currently open for the process.

correctly this is guaranteed as well?

Reply:
So the `fileno` check, unlike `remove` etc, seems to consistently work on unix and windows (as in: removing gc.colect = failure) si I'd say we go with that.

Reply:
That sounds pretty good, though do you know if that's reflecting the POSIX standard or just implementation detail? It might fail on one of the supported POSIX platforms?

The ability to remove file while open is a Linux thing I'd say, not POSIX, so not necessarily portable for all platforms supported by this interface.

Reply:
It appears here: https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_14 so it's the standard?

Reply:
Regardless it's a novel solution, providing something to check at least, I haven't thought of anything better.

You could possibly grab the fileno while it's open, then use ffillib to hit the C functions directly to check if it's still open... but that's possibly more effort than is with and adding extra dependencies to the test

Reply:
Yeah but the windows port doesn't do ffilib so that's not really an option

Reply:
Ah yes of course - thanks for that link I went looking for a POSIX reference for it myself quickly but didn't hit anything that detailed.

In that case this definitely seems like an elegant solution to me too!